### DISCUSSION TITLE
How can you save the state of number boxes?

### DISCUSSION THREAD

**Discussion by unfamiliartunes** [pierreguillot/Camomile#259] [Q&A] [answered]
**Hi!**
I just started learning PureData and Camomile last month. I'm creating a spectral delay/filter plugin, and one of the features allows for the changing of the delay time with a number box. However, in order for this to be practical, the delay time you set (anywhere from 10 to 10,000 ms) needs to be saved into the preset/state. I've figured out how to do this with arrays (in the documentation), but number boxes escape me. Can anyone help? Thank you! (I am using Camomile v1.0.7)

**Comment by pierreguillot** [comment, marked as answer]
Hi,
An easy solution would be to use a plugin parameter, the value will be automatically saved and restored. Otherwise, if you don't want the [nbx] object to be linked to a parameter, you can save the object with [receive save] [bang] -> [nbx]-> [save myvalue $1( -> [send camomile] and restore object with [receive load] -> [route myvalue] -> [nbx], of course you can replace "myvalue" with whatever you want and you can use the same approach for many [nbx] objects (if each one uses a specific symbol instead of "myvalue").

**Reply by unfamiliartunes** [reply]
Thanks! If I were you, I'd implement the code for this (and other parameters) in the wiki! 👍

**Reply by pierreguillot** [reply]
I tried to put all the generic information about the parameters and programs mechanisms in the wiki. Of course, I cannot cover all possible implementations. But if you think that some parts could be more detailed, do not hesitate to make your suggestion, the wiki is collaborative ;)

**Reply by pierreguillot** [reply]
Hre is a solution:  
<img width="178" alt="camomile-save-load-value" src="https://user-images.githubusercontent.com/1409918/117535879-39b75b00-aff8-11eb-800a-0260ba97a132.png">

Tip: If you need to debug your patch, you can use the [print] object and see the output in the console.

**Reply by unfamiliartunes** [reply]
This isn't working for some reason. I'm beginner and I'm probably getting something wrong, what exactly do I replace "myvalue" with, and where do I route it?

**Reply by pierreguillot** [reply]
Can you share your plugin/patch ? And give further information about what you do (your OS/DAW/etc.) ? It will be simpler to figure out the problem.

**Reply by pierreguillot** [reply]
As you deleted the last two posts and asked another question, I guess this issue is resolved ;)

**Reply by unfamiliartunes** [reply]
> As you deleted the last two posts and asked another question, I guess this issue is resolved ;)

You'd be right! I found a way to link arrays to atom boxes thru messaging. It might not be the prettiest way, but it works and allows for "default" values as well!
